src/daemon/rpmostreed-transaction-types.c

```diff
@@ -2060,6 +2060,158 @@ rpmostreed_transaction_new_refresh_md (GDBusMethodInvocation *invocation,
 
 }
 
+/* ================================ ModifyYumRepo ================================ */
+
+typedef struct {
+  RpmostreedTransaction parent;
+  char *osname;
+  char *repo_id;
+  GVariant *repo_options;
+} ModifyYumRepoTransaction;
+
+typedef RpmostreedTransactionClass ModifyYumRepoTransactionClass;
+
+GType modify_yum_repo_transaction_get_type (void);
+
+G_DEFINE_TYPE (ModifyYumRepoTransaction,
+               modify_yum_repo_transaction,
+               RPMOSTREED_TYPE_TRANSACTION)
+
+static void
+modify_yum_repo_transaction_finalize (GObject *object)
+{
+  ModifyYumRepoTransaction *self;
+
+  self = (ModifyYumRepoTransaction *) object;
+  g_free (self->osname);
+  g_free (self->repo_id);
+  g_variant_unref (self->repo_options);
+
+  G_OBJECT_CLASS (modify_yum_repo_transaction_parent_class)->finalize (object);
+}
+
+static DnfRepo *
+get_dnf_repo_by_id (RpmOstreeContext *ctx, const char *repo_id)
+{
+  DnfContext *dnfctx = rpmostree_context_get_dnf (ctx);
+
+  GPtrArray *repos = dnf_context_get_repos (dnfctx);
+  for (guint i = 0; i < repos->len; i++)
+    {
+      DnfRepo *repo = repos->pdata[i];
+      if (g_strcmp0 (dnf_repo_get_id (repo), repo_id) == 0)
+        return repo;
+    }
+
+  return NULL;
+}
+
+static gboolean
+modify_yum_repo_transaction_execute (RpmostreedTransaction *transaction,
+                                     GCancellable *cancellable,
+                                     GError **error)
+{
+  ModifyYumRepoTransaction *self = (ModifyYumRepoTransaction *) transaction;
+  OstreeSysroot *sysroot = rpmostreed_transaction_get_sysroot (transaction);
+
+  if (self->repo_id == NULL)
+    return glnx_throw (error, "Invalid repo_id");
+
+  if (self->repo_options == NULL)
+    return glnx_throw (error, "Invalid repo_options");
+
+  rpmostree_transaction_set_title ((RPMOSTreeTransaction*)self, "modify-yum-repo");
+
+  g_autoptr(OstreeDeployment) cfg_merge_deployment =
+    ostree_sysroot_get_merge_deployment (sysroot, self->osname);
+  g_autoptr(OstreeDeployment) origin_merge_deployment =
+    rpmostree_syscore_get_origin_merge_deployment (sysroot, self->osname);
+
+  /* but set the source root to be the origin merge deployment's so we pick up releasever */
+  g_autofree char *origin_deployment_root =
+    rpmostree_get_deployment_root (sysroot, origin_merge_deployment);
+
+  OstreeRepo *ot_repo = ostree_sysroot_repo (sysroot);
+  g_autoptr(RpmOstreeContext) ctx = rpmostree_context_new_system (ot_repo, cancellable, error);
+
+  /* We could bypass rpmostree_context_setup() here and call dnf_context_setup() ourselves
+   * since we're not actually going to perform any installation. Though it does provide us
+   * with the right semantics for install/source_root. */
+  if (!rpmostree_context_setup (ctx, NULL, origin_deployment_root, NULL, cancellable, error))
```

Comment:
I think we can avoid doing fetching the `origin_merge_deployment` and just provide `NULL` for `origin_deployment_root` here (since in this case, I don't think we actually need libdnf to know the right `$releasever` var in order to modify repo configs).

Reply:
Removed those.